security/cmd/node_agent_k8s/main.go

```diff
@@ -128,78 +127,31 @@ var (
 	// rootCmd defines the command for node agent.
 	rootCmd = &cobra.Command{
 		Use:   "nodeagent",
-		Short: "Node agent",
 		RunE: func(c *cobra.Command, args []string) error {
 			if err := log.Configure(loggingOptions); err != nil {
 				return err
 			}
 
 			applyEnvVars(c)
-
 			gatewaySdsCacheOptions = workloadSdsCacheOptions
 
 			if err := validateOptions(); err != nil {
 				return err
 			}
 
 			stop := make(chan struct{})
-
-			workloadSecretCache, gatewaySecretCache := newSecretCache(serverOptions)
-			if workloadSecretCache != nil {
```

Comment:
we need to keep this, move to ```server.Stop()``` ?

Reply:
Does it matter? `Stop` is never invoked in production usage. (only invoked here as defer) and my understanding is that WaitSignal will block always.

Stop is invoked as part of the secretcache_test.go code, which I understand because that can avoid the number of go routines increase too much.